stdlib/public/core/StringUnicodeScalarView.swift

```diff
@@ -581,13 +570,14 @@ extension String.UnicodeScalarIndex {
   public func samePosition(in characters: String) -> String.Index? {
     return String.Index(self, within: characters)
   }
+}
 
-  internal var _isOnGraphemeClusterBoundary: Bool {
-    let scalars = String.UnicodeScalarView(_core)
-    if self == scalars.startIndex || self == scalars.endIndex {
+extension String.UnicodeScalarView {
+  internal func _isOnGraphemeClusterBoundary(_ i: Index) -> Bool {
```

Comment:
Maybe I need to look in to it more, but is there a reason why the cluster boundary check is in UnicodeScalarView but the cluster measurement functions are in CharacterView?